src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -738,4 +735,148 @@ private String assignUserRole(Map<String, Object> requestBody, List<String> exis
+						props.getSbUrl() + props.getLmsUserSignUpPath(), reqObj, getDefaultHeaders());
+				if (Constants.OK.equalsIgnoreCase((String) readData.get(Constants.RESPONSE_CODE))) {
+					Map<String, Object> result = (Map<String, Object>) readData.get(Constants.RESULT);
+					String userId = (String) result.get(Constants.USER_ID);
+					request.put(Constants.USER_ID, userId);
+					Map<String, Object> userData = getUsersReadData(userId, StringUtils.EMPTY, StringUtils.EMPTY);
+					if (CollectionUtils.isNotEmpty(Collections.singleton(userData))) {
+						request.put(Constants.USER_NAME, userData.get(Constants.USER_NAME));
+
+						Map<String, Object> updateRequest = new HashMap<>();
+						Map<String, Object> updateRequestBody = new HashMap<String, Object>();
+						updateRequestBody.put(Constants.USER_ID, userId);
+						Map<String, Object> profileDetails = new HashMap<String, Object>();
+						profileDetails.put(Constants.MANDATORY_FIELDS_EXISTS, false);
+						Map<String, Object> employementDetails = new HashMap<String, Object>();
+						employementDetails.put(Constants.DEPARTMENTNAME, userData.get(Constants.ORG_NAME));
+						profileDetails.put(Constants.EMPLOYMENTDETAILS, employementDetails);
+						Map<String, Object> personalDetails = new HashMap<String, Object>();
+						personalDetails.put(Constants.FIRSTNAME.toLowerCase(), requestData.get(Constants.FIRST_NAME));
+						personalDetails.put(Constants.SURNAME, requestData.get(Constants.LAST_NAME));
+						personalDetails.put(Constants.PRIMARY_EMAIL, userData.get(Constants.EMAIL));
+						personalDetails.put(Constants.USER_NAME, userData.get(Constants.USER_NAME));
+						profileDetails.put(Constants.PERSONAL_DETAILS, personalDetails);
+
+						Map<String, Object> professionDetailObj = new HashMap<String, Object>();
+						professionDetailObj.put(Constants.ORGANIZATION_TYPE, Constants.GOVERNMENT);
+						if (StringUtils.isNotEmpty((String) userData.get(Constants.POSITION))) {
+							professionDetailObj.put(Constants.DESIGNATION, userData.get(Constants.POSITION));
+						}
+						List<Map<String, Object>> professionalDetailsList = new ArrayList<Map<String, Object>>();
+						professionalDetailsList.add(professionDetailObj);
+						profileDetails.put(Constants.PROFESSIONAL_DETAILS, professionalDetailsList);
+
+						updateRequestBody.put(Constants.PROFILE_DETAILS, profileDetails);
+						updateRequest.put(Constants.REQUEST, updateRequestBody);
+						Map<String, Object> updateReadData = (Map<String, Object>) outboundRequestHandlerService
+								.fetchResultUsingPatch(props.getSbUrl() + props.getLmsUserUpdatePath(), updateRequest, getDefaultHeaders());
+
+						if (Constants.OK.equalsIgnoreCase((String) updateReadData.get(Constants.RESPONSE_CODE))) {
+							Map<String,Object> roleMap=new HashMap<>();
+							roleMap.put(Constants.ORGANIZATION_ID,userData.get(Constants.ROOT_ORG_ID));
+							roleMap.put(Constants.USER_ID,userId);
+							retValue=assignRole(roleMap);
```

Comment:
use assign role in the user utility controller, we already have this method there

Reply:
assign role inside userUtility takes model class object , Here I am passing map thats why created one private method inside ProfileService